arbnode/dataposter/data_poster.go

```diff
@@ -389,13 +397,19 @@ func (p *DataPoster) feeAndTipCaps(ctx context.Context, nonce uint64, gasLimit u
 	}
 
 	elapsed := time.Since(dataCreatedAt)
-	// MaxFeeCap = (BacklogOfBatches^2 * UrgencyGWei^2 + TargetPriceGWei) * GWei
-	maxFeeCap :=
-		arbmath.FloatToBig(
-			(float64(arbmath.SquareUint(backlogOfBatches))*
-				arbmath.SquareFloat(config.UrgencyGwei) +
-				config.TargetPriceGwei) *
-				params.GWei)
+	parameters := make(map[string]interface{})
+	parameters["BacklogOfBatches"] = float64(backlogOfBatches)
+	parameters["UrgencyGWei"] = config.UrgencyGwei
+	parameters["ElapsedTime"] = float64(elapsed)
+	parameters["ElapsedTimeBase"] = float64(config.ElapsedTimeBase)
+	parameters["ElapsedTimeImportance"] = config.ElapsedTimeImportance
+	parameters["TargetPriceGWei"] = config.TargetPriceGwei
+	parameters["GWei"] = params.GWei
+	result, err := p.maxFeeCapExpression.Evaluate(parameters)
```

Comment:
nit: you can do 

params := map[string]any {
   "BacklogOfBatches": float64(backlogOfBatches),
    "UrgencyGWei": config.UrgencyGwei,
    "ElapsedTime": float64(elapsed),
    ....
}

Less of a boilerplate code that way.

Reply:
Also, I would factor this out into a helper function, something like this:
```
func (dp *DataPoster) evalFeeCapExpr(backlogOfBatches uint64) (*big.Int, error) {
	elapsed := time.Since(dataCreatedAt)		
	cfg := p.config()
	parameters = map[string]any {
 		"BacklogOfBatches": float64(backlogOfBatches),
		"UrgencyGWei",  cfg.UrgencyGwei,
		"ElapsedTime",  float64(elapsed),
		"ElapsedTimeBase",  float64(cfg.ElapsedTimeBase),
		"ElapsedTimeImportance",  cfg.ElapsedTimeImportance,
		"TargetPriceGWei",  cfg.TargetPriceGwei,
		"GWei",  params.GWei,
	}
	result, err := p.maxFeeCapExpression.Evaluate(parameters)
	if err != nil {
		return nil, nil, fmt.Errorf("error evaluating maxFeeCapExpression: %w", err)
	}
        res, ok := result.(float64)
        if !ok {
        return nil, fmt.Errorf("maxFeeCapExpression did not evaluate to float, result: %v\n", res)
        }
        return arbmath.FloatToBig(res), nil
}
```


That way it's cleaner and much easier to test this part separately too.

Reply:
fair point! I'll implement the logic in a helper function